Sources/SourceKitLSP/DocumentManager.swift

```diff
@@ -140,27 +141,25 @@ public final class DocumentManager {
   ///   - newVersion: The new version of the document. Must be greater than the
   ///     latest version of the document.
   ///   - edits: The edits to apply to the document
-  ///   - willEditDocument: Optional closure to call before each edit. Will be
-  ///     called multiple times if there are multiple edits.
+  ///   - requiresUpcomingEdits: Flag to return upcoming editst. Will be
+  ///     calling ``willEditDocument`` for each edit made
   /// - Returns: The snapshot of the document before the edit and the snapshot
-  ///   of the document after the edit.
+  ///   of the document after the edit, including all edits that were made.
```

Comment:
I would like to add whether the edits are concurrent or sequential to the doc comment. (see `ConcurrentEdits` in swift-syntax, it has some comments on the difference).

I would add 

```
The edits are sequential, ie. the edits are expected to be applied in order and later values in this array assume that previous edits are already applied.
```